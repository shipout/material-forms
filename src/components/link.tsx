import React from 'react'
import Button from '@material-ui/core/Button'
import { Link } from 'react-router-dom'
import { ObjectHelper } from '../object'

interface IProps {
  url: string
  className?: string
  contained?: boolean
  color?: 'default' | 'inherit' | 'primary' | 'secondary'
  rounded?: boolean
}

export class ButtonLink extends React.Component<IProps, any> {
  public render() {
    let color = this.props.color
    if (ObjectHelper.isEmpty(this.props.color)) {
      color = 'default'
    }

    return (
      <Button
        className={`button ${
          this.props.className ? this.props.className : ''
        } ${this.props.rounded ? '--rounded' : ''}`}
        variant={this.props.contained ? 'contained' : 'text'}
        color={color}
      >
        <Link to={this.props.url}>{this.props.children}</Link>
      </Button>
    )
  }
}
