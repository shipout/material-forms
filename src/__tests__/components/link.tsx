import React from 'react'
import { shallow } from 'enzyme'
import { ButtonLink } from '../../components/button/link'
import * as enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
enzyme.configure({ adapter: new Adapter() })

describe('ButtonLink', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<ButtonLink url='url'>text</ButtonLink>)
    expect(component).toMatchSnapshot()
  })
})
