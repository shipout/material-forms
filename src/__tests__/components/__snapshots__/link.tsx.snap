// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonLink should render correctly in "debug" mode 1`] = `
<WithStyles(ForwardRef(Button))
  className="button  "
  color="default"
  variant="text"
>
  <Link
    to="url"
  >
    text
  </Link>
</WithStyles(ForwardRef(Button))>
`;
